lib/cylc/task_state.py

```diff
@@ -360,7 +361,11 @@ def release(self):
         old_status = self._state_pre_hold
         self._state_pre_hold = None
         self.log(INFO, 'held => %s' % (old_status))
-        self.reset_state(old_status)
+
+        # Turn off submission and execution timeouts.
+        self.submission_timer_timeout = None
+        self.execution_timer_timeout = None
+        self.set_state(old_status)
```

Comment:
Actual fix here. I think we probably don't need to reset the timeouts, but it is harmless.